tutorials/tour/_posts/2017-02-13-classes.md

```diff
@@ -9,46 +9,105 @@ categories: tour
+    s"($x, $y)"
 }
 ```
 
-Classes in Scala are parameterized with constructor arguments. The code above defines two constructor arguments, `x` and `y`; they are both visible in the whole body of the class.
+This `Point` class has four members: the variables `x` and `y` and the methods `move` and
+`toString`. Unlike many other languages, the primary constructor is in the class signature `(var x: Int, var y: Int)`. The `move` method takes two integer arguments but does not return a value (the return type `Unit` corresponds to `void` in Java-like languages). `toString`, on the other hand, does not take any parameters but returns a `String` value. Since `toString` overrides the inherited `toString` method, it is tagged with the `override` keyword.
+
+## Constructors
+
+Constructors can have optional parameters by providing a default value like so:
 
-The class also includes two methods, `move` and `toString`. `move` takes two integer arguments but does not return a value (the return type `Unit` corresponds to `void` in Java-like languages). `toString`, on the other hand, does not take any parameters but returns a `String` value. Since `toString` overrides the pre-defined `toString` method, it is tagged with the `override` keyword.
+```tut
+class Point(var x: Int = 0, var y: Int = 0)
+
+val origin = new Point()
 
-Note that in Scala, it isn't necessary to say `return` in order to return a value. The value returned from a method is simply the last value in the method body. In the case of the `toString` method above, the expression after the equals sign is evaluated and returned to the caller.
+println(origin) // (0, 0)
+```
 
-Classes are instantiated with the `new` primitive, as follows:
+Optional parameters give you a lot of flexibility but if you still need more flexibility, you can create auxiliary constructors which are procedures (i.e. a `def` with no `=` and no return type).
 
 ```tut
-object Classes {
-  def main(args: Array[String]) {
-    val pt = new Point(1, 2)
-    println(pt)
-    pt.move(10, 10)
-    println(pt)
+class Point {
+  def this(x: Int, y: Int) {
+    this()
+    if (x > 100 || y > 100) println("Warning: Out of bounds")
   }
 }
 ```
+There are few requirements for auxiliary constructors:
+* They must start with `def this`
+* The parameters must _not_ be declared `val` or `var`
+* The first statement in the body must call the primary constructor or another auxiliary constructor. In the example above, the implicit primary constructor, `this()`, is called.
+
+## Member scope
+Methods are public by default. Use the `private` access modifier
+to hide them from outside of the class.
+```tut
+import scala.math._
+
+class Point(var x: Double, var y: Double) {
+  private def distanceFromOrigin = sqrt(x * x + y * y)
+
+  override def toString: String =
+    s"Point: ($x, $y), Distance from origin: $distanceFromOrigin"
+  }
+```
+
+You can also simply nest private methods inside of the public ones
+that use them:
```

Comment:
I think this is a general point related to scopes more than classes. I’m not sure it’s best place is in this page.